front/components/vaults/VaultAppsList.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Chip,
+  CommandLineIcon,
+  DataTable,
+  Searchbar,
+  Spinner,
+} from "@dust-tt/sparkle";
+import type {
+  ConnectorType,
+  DataSourceViewCategory,
+  PlanType,
+  VaultType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import type { CellContext } from "@tanstack/react-table";
+import type { ComponentType } from "react";
+import { useRef } from "react";
+import { useState } from "react";
+import * as React from "react";
+
+import { useApps } from "@app/lib/swr";
+import { classNames } from "@app/lib/utils";
+
+type RowData = {
+  category: string;
+  label: string;
+  icon: ComponentType;
+  connector?: ConnectorType;
+  fetchConnectorError?: string;
+  visibility: string;
+  workspaceId: string;
+  onClick?: () => void;
+};
+
+type Info = CellContext<RowData, unknown>;
+
+type VaultResourcesListProps = {
+  owner: WorkspaceType;
+  plan: PlanType;
+  isAdmin: boolean;
+  vault: VaultType;
+  systemVault: VaultType;
+  category: DataSourceViewCategory;
+  onSelect: (sId: string) => void;
+};
+
+const getTableColumns = () => {
+  return [
+    {
+      header: "Name",
+      accessorKey: "label",
+      id: "label",
+      cell: (info: Info) => (
+        <DataTable.CellContent icon={info.row.original.icon}>
+          <span className="font-bold">{info.row.original.label}</span>
+        </DataTable.CellContent>
+      ),
+    },
+    {
+      header: "Visibility",
+      cell: (info: Info) => (
```

Comment:
same here, use CellContext<RowData,AppVisibility> , add an accessorKey `visibility`, and use `getValue()` instead of `info.row.original.visibility`. HAving a proper accessor and value type is required if we want to be able to faceting on this column.